src/hardware/siglent-sds/protocol.c

```diff
@@ -167,11 +167,18 @@ SR_PRIV int siglent_sds_capture_start(const struct sr_dev_inst *sdi)
 				devc->num_frames + 1, devc->limit_frames);
 			if (siglent_sds_config_set(sdi, "ARM") != SR_OK)
 				return SR_ERR;
+			/* This pause is necessary for large memory depths. */
+			struct sr_channel *ch = devc->channel_entry->data;
+			if (ch->type == SR_CHANNEL_ANALOG) {
+				float wait = devc->memory_depth_analog / 100;
+				sr_spew("Waiting %.f ms for the instrument to enter ARM mode.", wait / 1000);
+				g_usleep(wait);
+			}
 			if (sr_scpi_get_string(sdi->conn, ":INR?", &buf) != SR_OK)
 				return SR_ERR;
 			sr_atoi(buf, &out);
 			g_free(buf);
-			if (out == DEVICE_STATE_TRIG_RDY) {
+			if (out == DEVICE_STATE_TRIG_RDY || out == DEVICE_STATE_STOPPED) {
```

Comment:
another change that impacts other devices, would like to hear rationale, testing on other models to confirm no breakage, and separate commit unless it's directly related to sds2000 support